src/rvary_fmt_plug.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * Cracker for RVARY STATUS and RVARY SWITCH commands of RACF
+ *
+ * This software is Copyright (c) 2020 Davide Girardi <davidegirardi smtpgoogle>,
+ * and it is hereby released to the general public under the following terms:
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Dump format: $rvary$**hash
+ *
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_rvary;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_rvary);
+#else
+
+#include <string.h>
+#include <openssl/des.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "arch.h"
+#include "crc32.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+
+#define FORMAT_LABEL            "RVARY"
+#define FORMAT_NAME             ""
+#define FORMAT_TAG              "$rvary$*"
+#define FORMAT_TAG_LEN          (sizeof(FORMAT_TAG)-1)
+#define ALGORITHM_NAME          "DES 32/" ARCH_BITS_STR
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        7
+#define PLAINTEXT_LENGTH        8
+#define CIPHERTEXT_LENGTH       16
+#define BINARY_SIZE             8
+#define SALT_SIZE               0
+#define BINARY_ALIGN            sizeof(uint32_t)
+#define SALT_ALIGN              1
+
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      256
+
+#ifndef OMP_SCALE
+#define OMP_SCALE               4 // Tuned w/ MKPC for super
+#endif
+
+static const unsigned char a2e[256] = {
+	0,  1,  2,  3, 55, 45, 46, 47, 22,  5, 37, 11, 12, 13, 14, 15,
+	16, 17, 18, 19, 60, 61, 50, 38, 24, 25, 63, 39, 28, 29, 30, 31,
+	64, 79,127,123, 91,108, 80,125, 77, 93, 92, 78,107, 96, 75, 97,
+	240,241,242,243,244,245,246,247,248,249,122, 94, 76,126,110,111,
+	124,193,194,195,196,197,198,199,200,201,209,210,211,212,213,214,
+	215,216,217,226,227,228,229,230,231,232,233, 74,224, 90, 95,109,
+	121,129,130,131,132,133,134,135,136,137,145,146,147,148,149,150,
+	151,152,153,162,163,164,165,166,167,168,169,192,106,208,161,  7,
+	32, 33, 34, 35, 36, 21,  6, 23, 40, 41, 42, 43, 44,  9, 10, 27,
+	48, 49, 26, 51, 52, 53, 54,  8, 56, 57, 58, 59,  4, 20, 62,225,
+	65, 66, 67, 68, 69, 70, 71, 72, 73, 81, 82, 83, 84, 85, 86, 87,
+	88, 89, 98, 99,100,101,102,103,104,105,112,113,114,115,116,117,
+	118,119,120,128,138,139,140,141,142,143,144,154,155,156,157,158,
+	159,160,170,171,172,173,174,175,176,177,178,179,180,181,182,183,
+	184,185,186,187,188,189,190,191,202,203,204,205,206,207,218,219,
+	220,221,222,223,234,235,236,237,238,239,250,251,252,253,254,255
+};
+
+/* This is a2e[] with each entry XOR 0x55, left-shifted one bit
+   and finally with odd parity so that DES_set_key_unchecked
+   can be used directly.  This provides about 15% speed up.    */
```

Comment:
This code is an adaptation of the already present `racf_*_fmt_plug.c` files. The workaround comes from those. The same applies for all the other comments in this PR (which I am switching to WIP mode if I find how).

I started working on an alternative which supports more codepages (this shift is for the US codepage), I guess that using your bitslice DES is the way to go then.

Reply:
> The same applies for all the other comments in this PR

Maybe for most but not for all. I just took a look at `racf_fmt_plug.c` and it indeed shares much of this, but not all of the issues I pointed out are present there. For example, that format's use of `dirty` actually makes sense because that format is salted, and there's no inconsistent code indentation in that source file.

Reply:
The indentation issue is because I tried to keep the common lines untouched, to make diffing easier for you guys.
Another easy fix, just a `=G` and maybe a `:retab`.

Thank you for taking the time to guide me.

Reply:
I think we should address this when start using the bitslice approach. That will also include refactoring the current RACF and RACF KDFAES code.